### PR TITLE
Grep for installed AND held packages.

### PR DESCRIPTION
**Pull request by doertedev**
See: [Issue #19977](https://github.com/docker/docker/issues/19977) - apt marks held packages (packages not to touch on upgrade/autoremove), which are not matched by ^ii.

### CODE REVIEW

File: hack/install.sh [hunk 1]
```diff
@@ -372,7 +372,7 @@ do_install() {
 
 			# aufs is preferred over devicemapper; try to ensure the driver is available.
 			if ! grep -q aufs /proc/filesystems && ! $sh_c 'modprobe aufs'; then
-				if uname -r | grep -q -- '-generic' && dpkg -l 'linux-image-*-generic' | grep -q '^ii' 2>/dev/null; then
+				if uname -r | grep -q -- '-generic' && dpkg -l 'linux-image-*-generic' | egrep '^ii|^hi' 2>/dev/null; then
```
Comment (tianon) [inline hack/install.sh]: I'm +1, but I think this should be `grep -qE` (so it stays quiet and so we get expanded correctly):

```
       In addition, the variant programs egrep  and  fgrep  are  the  same  as
       grep -E  and grep -F, respectively.  These variants are deprecated, but
       are provided for backward compatibility.
```

**Comment by doertedev** [comment]
:+1:  thanks didnt know egrep was deprecated!

**Comment by LK4D4** [comment]
LGTM
@tianon feel free to merge

**Comment by calavera** [comment]
LGTM

**Comment by tianon** [comment]
:+1:

**Comment by thaJeztah** [comment]
@jfrazelle do we need this cherry-picked in the release branch?

**Comment by jfrazelle** [comment]
Sure sgtm up to Tibor but should be safe

On Sunday, February 7, 2016, Sebastiaan van Stijn <notifications@github.com>
wrote:

> @jfrazelle <https://github.com/jfrazelle> do we need this cherry-picked
> in the release branch?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/20038#issuecomment-181145566>.
>


-- 


Jessie Frazelle
4096R / D4C4 DD60 0D66 F65A 8EFC  511E 18F3 685C 0022 BFF3
pgp.mit.edu <http://pgp.mit.edu/pks/lookup?op=get&search=0x18F3685C0022BFF3>

**Pull request merged**